packages/server/src/middleware/nuxt.js

```diff
@@ -113,9 +113,9 @@ const defaultPushAssets = (preloadFiles, shouldPush, publicPath, options) => {
 
     const { crossorigin } = options.build
     const cors = `${crossorigin ? ` crossorigin=${crossorigin};` : ''}`
-    const ref = modern ? 'modulepreload' : 'preload'
+    const rel = `${(modern && asType === 'script') ? 'module' : ''}preload`
```

Comment:
```suggestion
    const rel = modern && asType === 'script' ? 'modulepreload' : 'preload'
```

Reply:
BTW, could you add a comment here (describe that modulepreload only supports script-like `as` value) for others can have better understanding  ?
https://html.spec.whatwg.org/multipage/links.html#link-type-modulepreload

Reply:
Done